relayer/message_relayer.go

```diff
@@ -173,10 +263,20 @@ func (r *messageRelayer) createSignedMessage(requestID uint32) (*warp.Message, e
 	}
 
 	// Construct the request
-	req := msg.MessageSignatureRequest{
-		MessageID: r.warpMessage.ID(),
+
+	// Make sure to use the correct codec
+	var reqBytes []byte
+	if r.relayer.sourceSubnetID == constants.PrimaryNetworkID {
+		req := coreEthMsg.MessageSignatureRequest{
+			MessageID: r.warpMessage.ID(),
+		}
+		reqBytes, err = coreEthMsg.RequestToBytes(coreEthCodec, req)
+	} else {
+		req := msg.MessageSignatureRequest{
+			MessageID: r.warpMessage.ID(),
+		}
+		reqBytes, err = msg.RequestToBytes(codec, req)
 	}
-	reqBytes, err := msg.RequestToBytes(codec, req)
 	if err != nil {
 		r.logger.Error(
 			"Failed to marshal request bytes",
```

Comment:
nit: think might be better to log Warp message ID over destination blockchain ID